### DISCUSSION TITLE
New Feature ideas

### DISCUSSION THREAD

**Discussion by Avetaemia** [unchihugo/FluentFlyout#8] [General]
modernflyouts (MF) has some features that are keeping me for now

MF's duration/seek bar is nice. this is a good option, but it needs to be coded a bit better when you implement it.
MF's thumbnail-in-the-background (instead of just the thumbnail as a small panel) is easier on the eyes for me in most instances.
MF seems to show the icon of the player — you give text, and having BOTH icon and text (with options to toggle each) would be preferable.

i think your paneling of the thumbnails is better, but i do like the thumbnail in the back as well.

MF allows for itself to be clicked & dragged for repositioning.

one feature that I would like handled (that MF struggles with) is multi-playback. sometimes i listen to multiple things at once, or i start another thing before I've stopped the initial playback. being able to neatly view & manage these audio sources would be a productivity changer.
i don't like having to find which of the dozens of tabs, or multiple windows i have open, has the currently playing audio that MF has lost because I started playing something else.

thank you Mr. H. Li, i hope to see your project grow and flourish. 

itemised:

— duration/seek bar, smart
— thumbnail-in-background (currently thumbnail-in-panel) ; options for each
— icon of the player (current text of the player) ; options for each
— text identifier of player should be Formalised. "Firefox", not "Mozilla.Firefox.App"
— click and drag to reposition
— multi-source identification
— modal-flyouts (hold ctrl and cycle volume up - delivers flyout of CPU/RAM/Temps; hold ctrl and cycle volume down - delivers flyout of area temperature, current weather status, etc) [low priority suggestion]

**Comment by unchihugo** [comment]
Hi, thank you for these suggestions! I'll definitely look into each of them.

I do like the idea of having the background show the thumbnail as well, but I think this will go against the design language (Fluent 2) a bit too much.

By the way, you mentioned that the seeker needs to be coded a bit better? Could you elaborate a bit on that? Thanks!

**Reply by Avetaemia** [reply]
MF gets confused sometimes. Sometimes he will know the duration, but not what is currently playing. At the moment, I am finding it identifies the current track's duration but does not specify where in the track it is playing. (He thinks it's constantly 00:00)

This is in contrast to KDE Connect, which can communicate the information properly (duration, and current position) to a separate device (tablet, phone). _[edit: changed "KDE" to "KDE Connect"]_

You've also pushed to developing "next track" info, which would be helpful information — this is obviously absent from MF (nice work). MF struggles to allow the next/previous track to be played, or a proper playback position to be selected. 

I don't know if these are issues because MF loses focus of what is playing, or because of age.

i think fxSound is open source, it might have aspects you'd find interesting (audio visualiser, change playback device) implementing in some way. 

the seek-bar as a visualiser is helpful to determine a track's activity (or lack thereof).

i might want to turn the volume up/down in the middle of the track. if the track is quiet when doing that, and the seek-bar shows the position is in the middle of the duration, then it would appear flat/still to depict that it is quiet.
or, perhaps the wrong playback device is selected, so I am experiencing no sound. I try to turn up the audio to hear it, and I see the seek-bar is bouncing, indicating that it _**is**_ playing audio to the audio device. so, I click on the flyout "change device" button.

**Comment by raspberryhead** [comment]
Plexamp support pls

**Reply by unchihugo** [reply]
Is Plexamp a media player on Windows? If so, they will have to integrate Windows media controls into their program if media controls do not work, which is out of our reach.

**Reply by GeneFJacket** [reply]
Yes, plexamp is the dedicated desktop app for plex music. As far as I'm aware it does have windows media controls built in.

**Comment by reu0str** [comment]
Is it possible to make it a little bigger? Or am I missing some way to do that. An option for like 3 different zoom level will be quite useful.

**Comment by WobblySausage** [comment]
Great program. I was wounding if there is a way to place the pop up on another screen. I quite like having it open, but not while gaming. It would be great if we could pick which screen to also assign it to

**Comment by DETBEW** [comment]


**Reply by DETBEW** [reply]
I often switch between multiple languages, so it’s important for me to clearly see when the keyboard layout changes. Many users have the same need, so adding a language-change flyout would be very useful.

**Comment by DorsalAxe** [comment]
Seeing as pressing the taskbar widget button opens the media flyout, would it be possible to make selecting it again close it? This would be useful, and I think would also better align the behaviour of the widget button with how native taskbar buttons/flyouts work.
